configs/milvus.yaml

```diff
@@ -173,11 +173,13 @@ dataCoord:
   segment:
     maxSize: 512 # Maximum size of a segment in MB
     sealProportion: 0.75 # It's the minimum proportion for a segment which can be sealed
-    assignmentExpiration: 2000 # The time of the assignment expiration in ms 
+    assignmentExpiration: 2000 # The time of the assignment expiration in ms
 
   compaction:
     enableAutoCompaction: true
+    enableAutoExpiration: false # Enable automatic entity expiration when doing compaction
     retentionDuration: 432000 # 5 days in seconds
+    entityExpiration: 259200 # 3 days in seconds, CAUTION expiration time is recommended over than a day
```

Comment:
I would say only introduce entityExpiration, and set default entityExpiration to max.

Also it would be super helpful if we enabled expire time setup for each collection

Reply:
Also we will keep data up to retentionDuration to support TimeTravel
We need to make sure entityExpiration >= retentionDuration otherwise time travel can not go back

Reply:
When it comes to expiration time setup for each collection, @sunby gave me related answer it in advance.

It's part of our conversation about it.
@Writtic : `entityExpiration` parameter for this feature will be used for datanode codes. But on `milvus.yaml`, the parameter group, `compaction`, isn't on datanode but datacoord like `dataCoord.compaction.enableAutoCompaction`. it will be fine where I create `entityExpiration` parameter on?

@sunby : In our design, DataNode only execute the command sent by DataCoord. So the compaction parameters were put under the `dataCoord` prefix. But I agree that we should extract them to a `compaction` group, or we will pass the parameters every time we send a compaction task. I can do this work after GA release.

Can I try to add this features to pass the parameter every time to compaction task and to setup expiration time for each collection?

cc: @sunby

Reply:
Um, and there is other problem. 

If I remove `CompactionEnableExpiration`, I should care about the condition like `CompactionEntityExpiration < math.MaxInt64` for preventing overflow and `timestamp.Add(time.Duration(CompactionEntityExpiration) * time.Second)`.

I can add the logic for checking those condition on `isExpiredEntity` method but it will get longer code. So, I wanna hear your opinion before getting to work.

Reply:
@xiaofan-luan If we want to enable these switch for eatch collection, we need to declare it when the collection is created. Can we implement it in another feature in which we need to change pymilvus and collection schema ?

Reply:
> When it comes to expiration time setup for each collection, @sunby gave me related answer it in advance.
> 
> It's part of our conversation about it. @Writtic : `entityExpiration` parameter for this feature will be used for datanode codes. But on `milvus.yaml`, the parameter group, `compaction`, isn't on datanode but datacoord like `dataCoord.compaction.enableAutoCompaction`. it will be fine where I create `entityExpiration` parameter on?
> 
> @sunby : In our design, DataNode only execute the command sent by DataCoord. So the compaction parameters were put under the `dataCoord` prefix. But I agree that we should extract them to a `compaction` group, or we will pass the parameters every time we send a compaction task. I can do this work after GA release.
> 
> Can I try to add this features to pass the parameter every time to compaction task and to setup expiration time for each collection?
> 
> cc: @sunby

I dont think it's a good idea to add this feature in this pr.

Reply:
> @xiaofan-luan If we want to enable this switch for eatch collection, we need to declare it when a collection is created. Can we implement it in another feature in which we need to change pymilvus and collection schema ?

sure, we will do it on another PR in the future